### PR TITLE
MAINTAINERS POC

### PR DESCRIPTION
**Pull request by JAremko**
Initial implementation for Spacemacs maintainers program from  #8686

### CODE REVIEW

**Comment by JAremko** [comment]
@syl20bnr  **Glorious Success!**  As usual :shipit:

File: MAINTAINERS.org [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+* Trailing white spaces
+
+| maintainers   |
+|---------------|
+| @robbyoconnor |
+| @robbyoconnor |
+| @robbyoconnor |
+| @robbyoconnor |
+| @robbyoconnor |
```
Comment (JAremko) [inline MAINTAINERS.org]: @robbyoconnor  suffer :shipit: / :hocho:

**Comment by JAremko** [comment]
So basically this thing will make sure that PR summons at least one maintainer from the [MAINTAINERS.org](https://github.com/JAremko/spacemacs-pr/blob/b2eccaee7faee6646010a8704998e1f552a80d9f/MAINTAINERS.org)

In time we can make it ask to summon not just a maintainer but a particular one based on what changes PR brings.

**Comment by JAremko** [comment]
Also mentioning teams and peoples in the commit messages will help tracking performance and stuff. Kinda useful side effect.

File: .travis-build.sh [hunk 2]
```diff
@@ -28,6 +28,32 @@ if  [ $TRAVIS_SECURE_ENV_VARS = false ] &&
     echo   "https://github.com/syl20bnr/spacemacs/blob/develop/CONTRIBUTING.org"
     exit 1
 fi
+# Make sure that a maintainer was summoned via commit message to review it
+if [ "${CHECK_MENTIONS}" = "true" ]; then
+	cd "${TRAVIS_BUILD_DIR}"
+	last_commit_message=$(git log -2 --pretty=%B)
+	if [[ ! "$last_commit_message" == *"@"* ]]; then
+		echo "Please summon a mainteiner"
```
Comment (swaroopch) [inline .travis-build.sh]: typo :) "maint*e*iner"

Comment (JAremko) [inline .travis-build.sh]: thx

File: MAINTAINERS.org (hunk 1 above)
Comment (robbyoconnor) [inline MAINTAINERS.org]: I hate you.

Comment (robbyoconnor) [inline MAINTAINERS.org]: @syl20bnr I forbid you to merge this! :hocho:

Comment (JAremko) [inline MAINTAINERS.org]: @robbyoconnor  I Shall Pass!!!

Comment (robbyoconnor) [inline MAINTAINERS.org]: FORBID!!! I FORBID! THIS IS NOT GONNA HAPPEN! :hocho: !!!! Also the gun needs to not be a water gun...damn it.

**Comment by JAremko** [comment]
OMG! That's so cool https://travis-ci.org/syl20bnr/spacemacs/jobs/224220429  :scream_cat:

**Comment by JAremko** [comment]
Btw I can add ` [skip travis]` to the commit message instead of constantly pining @robbyoconnor 
Could it be better :thinking:

**Comment by robbyoconnor** [comment]
> Btw I can add [skip travis] to the commit message instead of constantly pining @robbyoconnor
Could it be better :thinking:

That would only be logical, but you're not gonna do that.

**Comment by JAremko** [comment]
Apparently it notifies the mentioned person each time someone moves the commit around. So the bot solution is better after all (but will take some time to implement)

**Pull request closed without merging**